dependency_generator.py

```diff
@@ -107,106 +65,50 @@ def get_json():
-
-
-# Create ViewController.swift file for each of the module's views.
-def create_views(module):
-    directory = "{}/ViewControllers".format(module_directory(module))
-    for view in module.views:
-        file_path = "{}/{}{}ViewController.swift".format(directory,
-                                                         uppercase_first_letter(module.name),
-                                                         uppercase_first_letter(view))
-        with open(file_path, 'w+') as view_controller_file:
-            template = Template(templates.view_controller_template)
-            view_controller_file.write(template.render(upper_name=uppercase_first_letter(module.name),
-                                                       upper_view=uppercase_first_letter(view)))
-
-
-# Create App Dependencies File
-def create_app_dependencies(modules):
-    file_path = "{}/AppDependencies.swift".format(cwd())
-    with open(file_path, 'w+') as app_dependencies_file:
-        template = Template(templates.dependencies_template)
-        app_dependencies_file.write(template.render(lower_modules=lower_module_names(modules),
-                                                    upper_modules=upper_module_names(modules)))
-
-
-# Generate the Root Wire frame
-def create_root_wireframe():
-    create_generic_file("{}/Common".format(cwd()), "RootWireframe.swift", templates.root_wireframe)
-
-
-# create the storyboard file
-def create_storyboard(module):
-    create_generic_file(module_directory(module),
-                        "{}.storyboard".format(module.name.capitalize()),
-                        templates.storyboard)
-
-
 def main():
     check_valid_input()  # right now just checks number of arguments
     json_input = get_json()  # get json from system args -> exits if not valid JSON
+    project = get_project_from_json(json_input)  # Get project info
     all_modules = get_modules_from_json(json_input)
-    create_directories(all_modules)  # all needed directories
 
-    if "--existing" in sys.argv:
-        print_dependencies(all_modules)
-    else:
-        create_app_dependencies(all_modules)  # dependencies file
-        create_root_wireframe()  # root Wireframe
+    if project.platform == "android":
+        create_android_directories(all_modules)  # TODO: Merge with ios method
 
-    # Create All Module Files
-    for module in all_modules:
-        create_views(module)
-        create_presenter(module)
-        create_interactor(module)
-        create_wireframe(module)
-        create_storyboard(module)
-    # end
-    print("{} Files Created.".format(file_count(all_modules)))
+        if "--existing" in sys.argv:
+            # TODO: Currently no difference between new and existing projects
+            pass
 
+        # Create All Module Files
+        for module in all_modules:
+            create_android_module(project.name, module)
```

Comment:
Don't think it's worth doing now, but a lot of this code could be cleaned up if you created `Classes` or a pipeline where we had `iOSGenerator` and `AndroidGenerator` that both shared a `BaseGenerator`. Could maybe write a `TODO` or something in the `README` about it. This way you could then just say generator.build and it would go do all of the things and call the write methods that overwrote the base method.